File: jvm/core/main/scala/com/webank/eggroll/core/containers/meta/KillContainersRequest.scala
package com.webank.eggroll.core.containers.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Containers

import scala.collection.JavaConverters._
import scala.language.implicitConversions

case class KillContainersRequest(sessionId: String, containers: Array[Long])

object KillContainersRequest {
  implicit def deserialize(byteString: ByteString): KillContainersRequest = {
    val proto = Containers.KillContainersRequest.parseFrom(byteString)
    KillContainersRequest(proto.getSessionId, proto.getContainerIdsList.asScala.map(_.toLong)(collection.breakOut))
  }

  implicit def serialize(src: KillContainersRequest): Array[Byte] = {
    Containers.KillContainersRequest.newBuilder()
      .setSessionId(src.sessionId)
      .addAllContainerIds(src.containers.map(_.asInstanceOf[java.lang.Long]).toSeq.asJava)
      .build().toByteArray
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/pojo/KillContainersRequest.java
package com.eggroll.core.pojo;

import com.google.protobuf.ByteString;
import com.webank.eggroll.core.meta.Containers;
import lombok.Data;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

@Data
public class KillContainersRequest  implements RpcMessage{

    Logger log = LoggerFactory.getLogger(KillContainersRequest.class);

    private String sessionId;
    private List<Long> containers;

    @Override
    public byte[] serialize() {
                return Containers.KillContainersResponse.newBuilder()
                .setSessionId(this.getSessionId()).build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data) {

        try {
            Containers.KillContainersRequest proto = Containers.KillContainersRequest.parseFrom(data);
            this.setSessionId(proto.getSessionId());
            this.setContainers(proto.getContainerIdsList());
        }catch (Exception e){
            log.error("KillContainersRequest.deserialize() error :" ,e);
        }

    }

//    public KillContainersRequest deserialize(byte[] bytes){
//        KillContainersRequest killContainersRequest = new KillContainersRequest();
//        try {
//            Containers.KillContainersRequest proto = Containers.KillContainersRequest.parseFrom(bytes);
//            killContainersRequest.setSessionId(proto.getSessionId());
//            killContainersRequest.setContainers(proto.getContainerIdsList());
//        }catch (Exception e){
//            log.error("KillContainersRequest.deserialize() error :" ,e);
//        }
//        return killContainersRequest;
//    }
//
//    public byte[] serialize(){
//        return Containers.KillContainersResponse.newBuilder()
//                .setSessionId(this.getSessionId()).build().toByteArray();
//    }
}
